services/diversity/src/persistence/richness_nos.js

```diff
@@ -0,0 +1,306 @@
+const { areaTypeKeys } = require('../util/appropriate_keys');
+
+module.exports = (
+  db,
+  {
+    richnessNos,
+  },
+  logger,
+) => ({
+  /**
+   * Find the values for the total number of species in the given area
+   *
+   * @param {String} areaType area type
+   * @param {String | Number} areaId area id
+   *
+   * @returns {Object[]} Number of total inferred and observed species.
+   */
+  findTotalNumberOfSpecies: async (areaType, areaId) => {
+    try {
+      return db('richness_nos as rn')
+        .select(
+          db.raw('coalesce(rn.total_inf, 0) as inferred'),
+          db.raw('coalesce(rn.total_obs, 0) as observed'),
+          db.raw('coalesce(rnr.total_obs, 0) as region_observed'),
+          db.raw('coalesce(rnr.total_inf, 0) as region_inferred'),
+        )
+        .leftJoin('richness_nos_regions as rnr', 'rn.id_region', 'rnr.id_region')
+        .where({ 'rn.geofence_type': areaTypeKeys(areaType), 'rn.geofence_id': areaId });
+    } catch (e) {
+      logger.error(e.stack || e.Error || e.message || e);
+      throw new Error('Error getting data');
+    }
+  },
+
+  /**
+   * Get the thresholds for the total number of species in a given area type
+   *
+   * @param {String} areaType area type.
+   * @param {String | Number} areaId area id
+   *
+   * @returns {Object[]} Number of inferred and observed species for the desired group.
+   */
+  findThresholdsTotalNumberOfSpecies: async (areaType, areaId) => {
+    try {
+      return richnessNos.query()
+        .where({ geofence_type: areaTypeKeys(areaType), geofence_id: areaId })
```

Comment:
This query is wrong, it will just return the queried area values. We need it to return the minimum and maximum values of the same `areaType` and `id_region` as the queried area.

Reply:
And the same applies for all thresholds queries

Reply:
Thresholds queries were changed according to the requirements.